src/Illuminate/Queue/BeanstalkdQueue.php

```diff
@@ -148,7 +148,7 @@ public function deleteMessage($queue, $id)
      */
     public function getQueue($queue)
     {
-        return $queue ?: $this->default;
+        return $this->getQueuePrefix().($queue ?: $this->default);
```

Comment:
ok, i think i'm following you now. correct me if I'm wrong.  the code

```php
$this->getQueuePrefix().($queue ?: $this->default)
```

appears in 3 of the queue drivers, so you're suggesting adding a method in the parent to handle this?

I agree this would work. I don't know if I see a **huge** benefit adding it. Might just be an additional function/layer that adds to the complexity.  I'm gonna leave it as is for now, but try a PR if this gets merged.